js/gateio.js

```diff
@@ -603,6 +606,90 @@ module.exports = class gateio extends Exchange {
         return { 'url': url, 'method': method, 'body': body, 'headers': headers };
     }
 
+    async fetchTransactions (code = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        if (since !== undefined) {
+            request['start'] = since;
+        }
+        let response = this.privatePostDepositswithdrawals (this.extend (request, params));
+        for (let i = 0; i < response['deposits'].length; i++) {
+            response['deposits'][i]['type'] = 'deposit';
+        }
+        for (let i = 0; i < response['withdraws'].length; i++) {
+            response['withdraws'][i]['type'] = 'withdrawal';
+        }
+        let deposits = this.parseTransactions (response['deposits'], params);
```

Comment:
There's a small error in this place – parseTransactions has 4 arguments, not two, so we can't ignore currency `code`, `since` and `limit`.

So it has to be: `this.parseTransactions (array, currency, since, limit)` (it uses `currency`, not `code`).

Also, I suggest that we save one call by concatenating first, then parsing, like so:

```
    let response = this.privatePostDepositswithdrawals (this.extend (request, params));
    for (let i = 0; i < response['deposits'].length; i++) {
        response['deposits'][i]['type'] = 'deposit';
    }
    for (let i = 0; i < response['withdraws'].length; i++) {
        response['withdraws'][i]['type'] = 'withdrawal';
    }
    let deposits = this.parseTransactions (response['deposits'], params);
    let withdrawals = this.parseTransactions (response['withdraws'], params);
    let transactions = this.arrayConcat (deposits, withdrawals);
    return this.filterByCurrencySinceLimit (this.sortBy (transactions, 'timestamp'), code, since, limit);
```

↓

```
    let currency = undefined;
    if (code !== undefined) {
        currency = this.currency (code);
    }
    const response = this.privatePostDepositswithdrawals (this.extend (request, params));
    const deposits = this.safeValue (response, 'deposits', []);
    const withdrawals = this.safeValue (response, 'withdrawls', []);
    const transactions = this.arrayConcat (deposits, withdrawals);
    return this.parseTransactions (transactions, currency, since, limit);
```

And we can determine the `type` from the `id` (note the first letter of the `id` string), so we don't need to pass it from the outside.

Can you add the edits I mentioned above?

Reply:
thanks, will make the changes now

Reply:
Thanks for fixing it, i'll merge it shortly.